tinytuya/BulbDevice.py

```diff
@@ -520,31 +519,35 @@ def set_brightness(self, brightness, nowait=False):
         # Determine which mode bulb is in and adjust accordingly
         state = self.state()
         data = None
+        msg = 'set_brightness: '
 
         if "mode" in state:
             if state["mode"] == "white":
                 # for white mode use DPS for brightness
                 if not self.has_brightness:
                     log.debug("set_brightness: Device does not appear to support brightness.")
                     # return error_json(ERR_FUNCTION, "set_brightness: Device does not support brightness.")
-                payload = self.generate_payload(
-                    CONTROL, {self.DPS_INDEX_BRIGHTNESS[self.bulb_type]: brightness}
-                )
-                data = self._send_receive(payload, getresponse=(not nowait))
-
-            if state["mode"] == "colour":
+                data = self.set_value( self.DPS_INDEX_BRIGHTNESS[self.bulb_type], brightness, nowait=nowait )
+                msg += 'No repsonse from bulb.'
+            elif state["mode"] == "colour":
                 # for colour mode use hsv to increase brightness
                 if self.bulb_type == "A":
                     value = brightness / 255.0
                 else:
                     value = brightness / 1000.0
                 (h, s, v) = self.colour_hsv()
                 data = self.set_hsv(h, s, value, nowait=nowait)
+                msg += 'No repsonse from bulb.'
+            else:
+                msg += "Device mode is not 'white' or 'colour', cannot set brightness."
+        else:
+            msg += 'Unknown bulb state.'
 
         if data is not None or nowait is True:
             return data
         else:
-            return error_json(ERR_STATE, "set_brightness: Unknown bulb state.")
+            log.debug( msg )
+            return error_json(ERR_STATE, msg)

```

Comment:
I mentioned how we use python exceptions in some cases and error responses in other place. But thinking about it, it seems we do this:

* If user provides invalid parameters (values) - raise exception
* If buld responds with error or invalid conditions - respond with error_json()